packages/wonder-blocks-data/util/response-cache.test.js

```diff
@@ -111,117 +112,597 @@ describe("./response-cache.js", () => {
+                    type: "MY_HANDLER",
+                    shouldRefreshCache: () => false,
+                    fulfillRequest: jest.fn(),
+                    cache: customCache,
+                };
+
+                // Act
+                const result = cache.getEntry(fakeHandler, "options");
+
+                // Assert
+                expect(result).toStrictEqual({data: "data!"});
+                expect(customCache.store).toBeCalledWith(
+                    fakeHandler,
+                    "options",
+                    {data: "data!"},
+                );
+            });
+
+            it("CSR should remove framework entry when custom cache misses", () => {
+                // Arrange
+                jest.spyOn(Server, "isServerSide").mockReturnValue(false);
+                const cache = new ResponseCache({
+                    MY_HANDLER: {
+                        MY_KEY: {data: "data!"},
+                    },
+                });
+                const customCache = {
+                    store: jest.fn(),
+                    retrieve: jest.fn(),
+                    remove: jest.fn(),
+                    removeAll: jest.fn(),
+                };
+                const fakeHandler: IRequestHandler<string, string> = {
+                    getKey: () => "MY_KEY",
+                    type: "MY_HANDLER",
+                    shouldRefreshCache: () => false,
+                    fulfillRequest: jest.fn(),
+                    cache: customCache,
+                };
+
+                // Act
+                const result = cache.getEntry(fakeHandler, "options");
+
+                // Assert
+                expect(result).toStrictEqual({data: "data!"});
+                // TODO(somewhatabstract): Abstract away need to poke insides
+                expect(cache._cache["MY_HANDLER"]["MY_KEY"]).toBeUndefined();
+            });
+        });
+    });
+
+    describe("#remove", () => {
+        it("should return false if nothing was removed", () => {
             // Arrange
-            const cache = new ResponseCache();
+            const cache = new ResponseCache({
+                MY_HANDLER: {
+                    MY_OTHER_KEY: {data: "data!"},
+                },
+            });
+            const customCache = {
+                store: jest.fn(),
+                retrieve: jest.fn(),
+                remove: jest.fn(),
+                removeAll: jest.fn(),
+            };
             const fakeHandler: IRequestHandler<string, string> = {
                 getKey: () => "MY_KEY",
                 type: "MY_HANDLER",
                 shouldRefreshCache: () => false,
                 fulfillRequest: jest.fn(),
-                cache: null,
+                cache: customCache,
             };
 
             // Act
-            cache.cacheData(fakeHandler, "options", "data");
-            const result = cache.getEntry(fakeHandler, "options");
+            const result = cache.remove(fakeHandler, "optionsA");
```

Comment:
Is the use of `"optionsA"` vs `"options"` meaningful?

Reply:
Meaningful? The options (via the key generation method in the request handler) are how requests are differentiated. If two things shared the same key, they would share the same cache. I likely made a change here to ensure that was taken into account.